libraries/iio/iio.c

```diff
@@ -788,34 +760,17 @@ ssize_t iio_register(struct iio_desc *desc, struct iio_interface *iio_interface)
 ssize_t iio_unregister(struct iio_desc *desc,
 						struct iio_interface *iio_interface)
 {
-	struct iio_interfaces *interfaces;
-	int16_t i, deleted = 0;
-
-	interfaces = (struct iio_interfaces *)calloc(1, sizeof(struct iio_interfaces));
-	if (!interfaces)
-		return FAILURE;
-
-	interfaces->interfaces = (struct iio_interface **)calloc(
-					 iio_interfaces->num_interfaces - 1,
-					 sizeof(struct iio_interface*));
-	if (!interfaces->interfaces) {
-		free(interfaces);
-		return FAILURE;
-	}
+	struct iio_interface	*to_remove_interface;
 
-	for(i = 0; i < iio_interfaces->num_interfaces; i++) {
-		if (!strcmp(iio_interface->name, iio_interfaces->interfaces[i]->name)) {
-			iio_interface = iio_interfaces->interfaces[i];
-			deleted = 1;
-			continue;
-		}
-		interfaces->interfaces[i - deleted] = iio_interfaces->interfaces[i];
-	}
-
-	interfaces->num_interfaces = iio_interfaces->num_interfaces - 1;
-	free(iio_interfaces);
+	/* Get if the item is found, get will remove it from the list */
+	return list_get_find(desc->interfaces_list, (void **)&to_remove_interface,
```

Comment:
Probably you need to free some memory for `to_remove_interface`

Reply:
to_remove_interface is allocated in the main application. Should be free there.
Unregister should just remove the entry from the list. This is what get function do.